geth/jail/internal/loop/loop.go

```diff
@@ -40,6 +42,27 @@ type Loop struct {
 	lock  sync.RWMutex
 	tasks map[int64]Task
 	ready chan Task
+	state
+}
+
+// state encapsulates the loop's open and closedness. This is a boolean variable
+// indicating whether the loop is open for new tasks and a lock to control
+// access to this boolean.
+type state struct {
+	accept bool
+	lock   sync.RWMutex
```

Comment:
Even if the mutex in `Loop` is called `lock` the typical convention is a short `mu`. A good chance to introduce it here. 😉 

By the way, nice trick encapsulating `state` as own type with its own mutex.